src/server/utils/dates.py

```diff
@@ -153,18 +124,18 @@ def weeks_to_ranges(values: Sequence[Union[Tuple[int, int], int]]) -> Sequence[U
     for interval in intervals:
         # no overlap, append the interval
         # caveat: we subtract 1 from interval[0] so that contiguous intervals are considered overlapping. i.e. [1, 1], [2, 2] -> [1, 2]
-        if not merged or merged[-1][1] < interval[0] - 1:
+        if not merged or merged[-1][1] < decrement(interval[0]):
```

Comment:
```suggestion
        if not merged or merged[-1][1] < interval[0] - time_unit:
```

(continuation of suggestion above)